lib/main.js

```diff
@@ -80,18 +85,26 @@ function neostandard (options) {
     ...files && { files },
   }
 
+  const tsConfig = ts
+    ? typescriptify(jsConfigs, {
+      files: ['**/*.ts', ...filesTs || []],
+      ignores,
+      name: 'neostandard/ts',
+    })
+    : undefined
+
   return [
     // To make it a global ignore, "ignores" has to be the lone key of the config: https://eslint.org/docs/latest/use/configure/configuration-files#globally-ignoring-files-with-ignores
     ...ignores ? [{ ignores }] : [],
     ...isNonEmpty(meta) ? [{ name: 'neostandard/meta', ...meta }] : [],
     ...jsConfigs,
-    ...ts
-      ? [typescriptify(jsConfigs, {
-          files: ['**/*.ts', ...filesTs || []],
-          ignores,
-          name: 'neostandard/ts',
-        })]
-      : [],
+    ...tsConfig ? [tsConfig] : [],
+    ...reactify({
+      ignores,
+      tsConfig,
+      reactConfig: react,
+      requireReactImport,
+    }),
```

Comment:
The idea here is to stack configuration with override priority. 

The configuration is expected to be built in that order: 

```js
 ...react
      ? [reactify(jsConfigs, {
      files: ['**/*.{jsx,cjsx,mjsx}'],
      name: 'neostandard/react',
      react: {
          version: 'detect',
      },
      languageOptions: {
          parserOptions: {
               jsx: true
          },
      },
      })]
      : [],
 ...ts
      ? [typescriptify(jsConfigs, {
          files: ['**/*.ts', ...filesTs || []],
          ignores,
          name: 'neostandard/ts',
        })]
      : [],
```